components/time/src/ixdtf.rs

```diff
@@ -723,13 +710,26 @@ impl Time {
 
     fn try_from_ixdtf_record(ixdtf_record: &IxdtfParseRecord) -> Result<Self, ParseError> {
         let time_record = ixdtf_record.time.ok_or(ParseError::MissingFields)?;
-        let time = Self::try_new(
+        Self::try_from_time_record(&time_record)
+    }
+
+    fn try_from_time_record(time_record: &TimeRecord) -> Result<Self, ParseError> {
+        let nanosecond = time_record
+            .fraction
+            .map(|fraction| {
+                fraction
+                    .to_nanoseconds()
+                    .ok_or(ParseError::ExcessivePrecision)
```

Comment:
The `ExcessivePrecision` error could be remove here in favor of calling `to_truncated_nanoseconds` if that were preferred.

Would also remove the `transpose` below.